packages/py/msgpack/msgpack_wrap/__init__.py

```diff
@@ -0,0 +1,27 @@
+import msgpack
+from dataclasses import dataclass, is_dataclass
+
+
+def sanitize(v):
+    for key in v.keys():
+        if is_dataclass(v[key]):
+            v[key] = v[key].__dict__
```

Comment:
do not access `__dict__` directly.

use `asdict` function from the dataclassses package.

Reply:
You don't need to recurse when using asdict, it will take care of it for you

Reply:
I think we still want to have recursive here. The reason is that if the value is a dictionary and has nested instances of classes, we need to check those. Does this makes sense to you?

Reply:
Gotcha! It makes sense in that case but try to run this code in python3.10 and you will understand why it is bad idea to access __dict__ :)

```python
from dataclasses import dataclass

@dataclass
class Test1:
    a: str
    
@dataclass(slots=True)
class Test2:
    a: str
    
print(Test1(a="1"))
print(Test1(a="1").__dict__)

print(Test2(a="2"))
print(Test2(a="2").__dict__)

```

Reply:
Yah I removed the `__dict__` and we are now using `asdict` :smile: